_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+Let's dive into what Z-score normalization is, why it's important, and how it's being used in OpenSearch.
+
+## What is Z-Score Normalization?
+
+Z-score normalization, also known as standardization, is a method of scaling data using mean and standard deviation. The formula for calculating the Z-score is:
+Z = (X - μ) / σ
+Where:
+
+* X is the original value
+* μ is the mean of the population
+* σ is the standard deviation of the population
+
+## When to use Z Score?
+
+Considering your index’s structure can help you decide which one to choose since each has advantages of its own. If your documents are more similar to one another and the top-k results of a typical query return documents that are very similar to one another and clustered together within the index, as seen in the graph below, Min-Max may be a better option.
+
+![Image for min max distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost1.jpg){: .img-fluid}
+
+However, Z-Score is more suited if the results are more evenly distributed and have some characteristics of a normal distribution, as shown in the example below.
+
+![Image for zscore distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost2.jpg){: .img-fluid}
+
+The basic flow to use between min max and z score looks like below:
+
+![Image for flow](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost3.png){: .img-fluid}
+
+### How to use Z Score?
+
+To use z_score, create a search pipeline and specify `z_score` as the `technique`:
+
+```
+PUT /_search/pipeline/z_score-pipeline
+{
+    "description": "Zscore processor for hybrid search",
+    "phase_results_processors": [
+        {
+            "normalization-processor": {
+                "normalization": {
+                    "technique": "z_score"
+                },
+                "combination": {
+                    "technique": "arithmetic_mean"
+                }
+            }
+        }
+    ]
+}
+```
+
+Next, create a hybrid query and apply the pipeline to it
+
+```
+
+
+POST my_index/_search?search_pipeline=z_score-pipeline
+{
+   "query": {
+     "hybrid": [
+         {}, // First Query
+         {}, // Second Query
+         ... // Other Queries
+     ] 
+   }
+}
+```
+
+
+
+## Benchmarking Z Score performance
+
+Benchmark experiments were conducted using an OpenSearch cluster consisting of a single r6g.8xlarge instance as the coordinator node, along with three r6g.8xlarge instances as data nodes. To assess Z Score’s performance comprehensively, we measured three key metrics across four distinct datasets. For information about the datasets used, see [Datasets](https://public.ukp.informatik.tu-darmstadt.de/thakur/BEIR/datasets/).
```

Comment:
Can you change the ending of this paragraph, typically for blogs we try to put more info regarding datasets, like for instance query length avg/med, doc length avg/med, number of queries. Not all of that is present in beir repo. 
You can add a table just for those 4 you used for experiments, similar to how it's done in this blog https://opensearch.org/blog/semantic-science-benchmarks, and also keep the reference to beir repo for other details